packages/language/src/preprocessor/instruction-generator.ts

```diff
@@ -201,6 +204,29 @@ function generateTokenInstruction(
   };
 }
 
+const DefaultCode: inst.ExpressionInstruction = {
+  kind: inst.InstructionKind.Number,
+  value: "0",
+};
+
+function generateNoteInstruction(
+  node: ast.NoteDirective,
+): inst.NoteInstruction | undefined {
+  const message = generateExpressionInstruction(node.message);
+  const code = generateExpressionInstruction(node.code) || DefaultCode;
+  return message && node.noteToken
+    ? {
+        kind: inst.InstructionKind.Note,
+        noteToken: node.noteToken,
+        message: message || {
+          kind: inst.InstructionKind.String,
+          value: "",
+        },
```

Comment:
Minor: We already know that the message is truthy.
```suggestion
        message,
```